BlocBrowser/BlocBrowser/AwesomeFloatingToolbar.m

```diff
@@ -0,0 +1,163 @@
+            label.alpha = 0.25;
+            
+            NSUInteger currentTitleIndex = [self.currentTitles indexOfObject:currentTitle];
+            NSString *titleForThisLabel = [self.currentTitles objectAtIndex:currentTitleIndex];
+            
+            label.titleLabel.textAlignment = NSTextAlignmentCenter;
+            label.titleLabel.font = [UIFont systemFontOfSize:15];
+            label.titleLabel.textColor = [UIColor whiteColor];
+            [label setTitle:titleForThisLabel forState:UIControlStateNormal];
+            
+            [label addTarget:self action:@selector(buttonTapped:) forControlEvents:UIControlEventTouchUpInside];
+            
+            [labelArray addObject:label];
+        }
+        self.labels = labelArray;
+        [self rotateColorAndAssign];
+        
+        for (UIButton *label in self.labels) {
+            [self addSubview:label];
+        }
+        
+        // gesture recognizer setup
+        self.panGesture = [[UIPanGestureRecognizer alloc] initWithTarget:self action:@selector(panFired:)];
+        [self addGestureRecognizer:self.panGesture];
+        self.pinchGesture = [[UIPinchGestureRecognizer alloc] initWithTarget:self action:@selector(pinchFired:)];
+        [self addGestureRecognizer:self.pinchGesture];
+        self.longPressGesture = [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(longPressFired:)];
+        [self addGestureRecognizer:self.longPressGesture];
+    }
+    
+    return self;
+}
+
+- (void) layoutSubviews {
+    for (UILabel *label in self.labels) {
+        NSUInteger index = [self.labels indexOfObject:label];
+        
+        CGFloat labelHeight = CGRectGetHeight(self.bounds) / 2;
+        CGFloat labelWidth = CGRectGetWidth(self.bounds) / 2;
+        CGFloat labelX = 0, labelY = 0;
+        
+        if (index < 2) {
+            labelY = 0;
+        } else {
+            labelY = labelHeight;
+        }
+        
+        if (index % 2 == 0) {
+            labelX = 0;
+        } else {
+            labelX = labelWidth;
+        }
+        
+        label.frame = CGRectMake(labelX, labelY, labelWidth, labelHeight);
+    }
+}
+
+- (void) rotateColorAndAssign {
+    UIColor *firstColor = [self.colors firstObject];
+    [self.colors removeObject:firstColor];
+    [self.colors addObject:firstColor];
+    
+    NSUInteger index = 0;
+    for (UILabel *label in self.labels) {
+        label.backgroundColor = [self.colors objectAtIndex:index++];
+    }
+}
+
+#pragma mark - Button Enabling
+
+- (void) setEnabled: (BOOL) enabled forButtonWithTitle: (NSString *) title {
+    NSUInteger index = [self.currentTitles indexOfObject:title];
+    
+    if (index != NSNotFound) {
+        UILabel *label = [self.labels objectAtIndex:index];
+        label.userInteractionEnabled = enabled;
+        label.alpha = enabled ? 1.0 : 0.25;
+    }
+}
+
+#pragma mark - Gesture Handling
+
+- (void) panFired:(UIPanGestureRecognizer *)recognizer {
```

Comment:
So `UIGestureRecognizerStateRecognized` only fires once, when the gesture is recognized. Like in the code in the checkpoint, you should use `UIGestureRecognizerStateChanged` so that your code gets called more often, as the user continues to pinch.